packages/dds/map/src/directory.ts

```diff
@@ -2184,20 +2183,23 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 		local: boolean,
 		localOpMetadata: unknown,
 	): boolean {
-		if (this.pendingClearMessageIds.length > 0) {
+		const firstPendingClearMessageId = this.pendingClearMessageIds[0];
+		if (firstPendingClearMessageId !== undefined) {
 			if (local) {
 				assert(
 					localOpMetadata !== undefined &&
 						isKeyEditLocalOpMetadata(localOpMetadata) &&
-						localOpMetadata.pendingMessageId < this.pendingClearMessageIds[0],
+						localOpMetadata.pendingMessageId < firstPendingClearMessageId,
 					0x010 /* "Received out of order storage op when there is an unackd clear message" */,
 				);
 				// Remove all pendingMessageIds lower than first pendingClearMessageId.
-				const lowestPendingClearMessageId = this.pendingClearMessageIds[0];
+				const lowestPendingClearMessageId = firstPendingClearMessageId;
 				const pendingKeyMessageIdArray = this.pendingKeys.get(op.key);
 				if (pendingKeyMessageIdArray !== undefined) {
 					let index = 0;
-					while (pendingKeyMessageIdArray[index] < lowestPendingClearMessageId) {
+					// Non null guarenteed by the way index is computed
+					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+					while (pendingKeyMessageIdArray[index]! < lowestPendingClearMessageId) {
```

Comment:
I think this is not accurate. `index` is being "conjured out of thin air" with an initial value of 0, and nothing guarantees that `pendingKeyMessageIdArray` has any items in it. A couple of checks in the console suggest to me that at runtime this behaves correctly even in that case (maybe a bit by luck?), but maybe a better solution here would be to start by assigning `pendingKeyMessageIdArray[index]` to a local variable, then `while (<variable> !== undefined && <variable> < lowestPendingClearMessageId) { index += 1; <variable> = pendingKeyMessageIdArray[index]; }`.